homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -75,51 +82,97 @@ async def async_setup_entry(
 class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
+    _attr_code_arm_required = False
+    _attr_supported_features = SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
+
     def __init__(self, coordinator: YaleDataUpdateCoordinator) -> None:
         """Initialize the Yale Alarm Device."""
         super().__init__(coordinator)
-        self._attr_name: str = coordinator.entry.data[CONF_NAME]
+        self._coordinator = coordinator
+        self._attr_name = coordinator.entry.data[CONF_NAME]
         self._attr_unique_id = coordinator.entry.entry_id
-        self._identifier: str = coordinator.entry.data[CONF_USERNAME]
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        """Return device information about this entity."""
-        return DeviceInfo(
-            identifiers={(DOMAIN, self._identifier)},
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.entry.data[CONF_USERNAME])},
             manufacturer=MANUFACTURER,
             model=MODEL,
-            name=str(self.name),
+            name=self._attr_name,
         )
 
-    @property
-    def state(self):
-        """Return the state of the device."""
-        return STATE_MAP.get(self.coordinator.data["alarm"])
-
-    @property
-    def available(self):
-        """Return if entity is available."""
-        return STATE_MAP.get(self.coordinator.data["alarm"]) is not None
-
-    @property
-    def code_arm_required(self):
-        """Whether the code is required for arm actions."""
-        return False
-
-    @property
-    def supported_features(self) -> int:
-        """Return the list of supported features."""
-        return SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
-
-    def alarm_disarm(self, code=None):
+    async def async_alarm_disarm(self, code=None) -> None:
         """Send disarm command."""
-        self.coordinator.yale.disarm()
-
-    def alarm_arm_home(self, code=None):
+        if self._coordinator.yale:
```

Comment:
I was reading the code in the coordinator and it looks like `yale` is always set given that the call to `get_updates` must succeed before the integration is loaded.

Can this check be made unnecessary? It seems like moving the `YaleSmartAlarmClient` init into the integration setup, and then passing that into the data update coordinator would be the most common way I see this type of thing handled.

Also could be handled in a followup... but it seems like this check isn't necessary otherwise

Reply:
It's only there to make mypy happy. If the integration can not load yale the coordinator will raise an error long before this check is even used.

Reply:
Yes I know why you added it. I'm asking about a better fix to remove the need for this. This is here because of where yeah is initialized and it can be simplified with what was suggested.

Reply:
I'm also curious what mypy thinks about referencing alarm_state when it hasn't been set I'm the "else" case. It doesn't look like mypy is enabled for this file yet, and I don't see that changing with this PR

Reply:
There is a few things that should be done which is out of scope of this PR. For example what you mention is one "problem" and another one is that the upstream package does auth on loading (hence the solution in place at the moment).
There will be further cleanups but it's out of scope of this PR.

Reply:
I'm all for doing things incrementally.  So then revert these three checks? I don't think it's these are the right mypy fixes for the reasons mentioned.

Reply:
I'd do it like this instead:
```py
if TYPE_CHECKING:
    assert self._coordinator.yale
...
```
This avoids it looking like a potential `NameError` below for `alarm_state`.